gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/AliasBasedTokenStateService.java

```diff
@@ -125,11 +126,27 @@ protected boolean isUnknown(final String token) {
     return isUnknown;
   }
 
+  @Override
+  protected void removeRevokedExpiredToken(final String token) {
+    if (isUnknown(token)) {
```

Comment:
Would it be better to employ validateToken() here?

Reply:
Addressed in new update.